IPython/lib/inputhook.py

```diff
@@ -177,13 +197,17 @@ def clear_app_refs(self, gui=None):
             the app for that toolkit.  References are not held for gtk or tk
             as those toolkits don't have the notion of an app.
         """
+        warn("`clear_app_refs` is deprecated since IPython 5.0 and will be removed in future versions.",
```

Comment:
I forgot that we instantiate it inside this module. Maybe we should just warn when the module is imported. Or is that annoying for tests?